build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,484 @@
+// This file is based on the Jenkins declarative pipeline syntax (as opposed to the scripted pipeline syntax)
+// https://jenkins.io/doc/book/pipeline/syntax/
+// https://jenkins.io/blog/2017/02/03/declarative-pipeline-ga/
+
+def botLabel = 'xamarin-android-master-pipeline-eval'
+
+def XADir = "xamarin-android"
+
+def publishBuildFilePaths = null
+def publishTestFilePaths = null
+
+def MSBUILD_AUTOPROVISION_ARGS="/p:AutoProvision=True /p:AutoProvisionUsesSudo=True /p:IgnoreMaxMonoVersion=False"
+
+def buildStatusProcessed = false
+def buildStatusUploaded = false
+
+def isPR = false                // Default to CI
+
+def processBuildStatus() {
+    def status = 0
+
+    try {
+        echo "processing build status"
+        sh "make package-build-status CONFIGURATION=${env.BuildFlavor}"
+    } catch (error) {
+        echo "ERROR : NON-FATAL : processBuildStatus: Unexpected error: ${error}"
+        status = 1
+    }
+
+    return status
+}
+
+def publishPackages(filePaths) {
+    def status = 0
+    try {
+        echo "Publishing to Azure storage account: https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/7b4817ae-218f-464a-bab1-a9df2d99e1e5/resourceGroups/Group/providers/Microsoft.ClassicStorage/storageAccounts/xamjenkinsartifact/overview"
```

Comment:
I wouldn't hardcode the URL to the storage container on the Azure Portal here. Regular developers won't have access to it anyway.

Reply:
Fixed with commit https://github.com/xamarin/xamarin-android/commit/802477c5ec36f98233e8a1b0eb0f05808e8151de